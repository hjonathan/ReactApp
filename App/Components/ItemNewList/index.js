import React, {Component} from "react";
import {View, Text, Image, TouchableOpacity} from "react-native";
import Icon from "react-native-vector-icons/FontAwesome";
import * as Res from "../../Assets/resources";
import Styles from "./styles";
import actions from "../../Actions";
import I18n from "../../I18n";
import Utils from "../../Utils";

/**
 * Class ItemNewList
 * Example:
 * import ItemNewList from '../ItemNewList';
 * render() {
 *      return(
 *          <ItemNewList data={object}/>
 *      ),
 * }
 */
class ItemNewList extends Component {
    /**
     * Constructor ItemList
     * @param props
     */
    constructor (props) {
        super(props);
        this.state = {
            data: (props && props.data) || {
                autoRoot: "FALSE",
                forms: [{
                    description: "",
                    formId: "",
                    formUpdateDate: "2018-01-18T00:23:52+00:00",
                    index: 1,
                    stepCondition: "",
                    stepId: "",
                    stepMode: "",
                    stepPosition: 1,
                    stepUidObj: "",
                    title: "",
                    triggers: {before: false, after: false}
                }
                ],
                offlineEnabled: "",
                processId: "",
                taskId: "",
                text: ""
            },
            progress: "",
            disableItemList: false
        };
    }

    /**
     * Component will receive props
     */
    componentWillReceiveProps (nextProps) {
        // We compare nextProps.parent.disableItemList with this.props.parent.disableItemList
        // and if both are different then we perform some operation, setState method
        if (nextProps.parent.disableItemList !== this.props.parent.disableItemList) {
            // Perform some operation
            this.setState({
                disableItemList: nextProps.parent.disableItemList
            });
        }
    }

    /**
     * Routes to formRender
     */
    goToFormRender () {
        if (!this.props.isConnected && !this.state.data.taskOffline) {
            Utils.showToast({message: I18n.t("you_need_Internet_connection_to_access_this_case")});
            return;
        }

        this.setState({
            disableItemList: true
        });

        this.props.parent.navigation.dispatch(actions.formRender.newCase({
            prj_uid: this.state.data.processId,
            act_uid: this.state.data.taskId,
            initTitle: this.state.data.forms[0].title
        }));
    }

    /**
     * Renders mark available offline
     * @returns {*}
     */
    renderOffline () {
        return (
            <View style={Styles.availableOffline}>
                <Icon name="check-circle-o" style={{color: Res.colors.successColor, fontSize: 24}} />
                <Text style={Styles.textOffline}>
                    {I18n.t("saved_on_Draft_new_activity_string")}
                </Text>
            </View>
        );
    }

    /**
     * Render
     * @returns {*}
     */
    render () {
        return (
            <TouchableOpacity onPress={() => this.goToFormRender()} disabled={this.state.disableItemList}>
                <View style={Styles.mainTopContainer} elevation={3}>
                    <View style={Styles.mainContainer}>
                        <View style={Styles.iconSection}>
                            <Image source={Res.images.newCaseIcon} style={Styles.circleImage} />
                        </View>
                        <View style={Styles.textSection}>
                            <Text style={Styles.textNewCase}>
                                {this.state.data.text}
                            </Text>
                            {this.state.data.taskOffline && this.renderOffline()}
                        </View>
                    </View>
                </View>
            </TouchableOpacity>
        );
    }
}

export default ItemNewList;
