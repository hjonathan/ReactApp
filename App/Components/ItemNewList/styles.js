import {StyleSheet} from "react-native";
import * as Res from "../../Assets/resources";

export default StyleSheet.create({
    mainContainer: {
        flexDirection: "row",
        backgroundColor: "#FFF",
        borderRadius: 6,
        overflow: "hidden",
        alignItems: "center"
    },
    iconSection: {
        flexDirection: "row",
        alignItems: "center",
        paddingHorizontal: 10
    },
    textSection: {
        paddingVertical: 13
    },
    circleImage: {
        width: 36,
        height: 36,
        resizeMode: "contain"
    },
    circleImageContainer: {
        backgroundColor: Res.colors.mainColor,
        borderRadius: 50,
        width: 36,
        height: 36
    },
    textNewCase: {
        color: "#000",
        paddingBottom: 3
    },
    controlGroup: {
        flex: 1,
        alignContent: "flex-end"
    },
    controls: {
        flexDirection: "row",
        justifyContent: "flex-end",
        paddingTop: 1
    },
    mainTopContainer: {
        flex: 1,
        margin: 3,
        borderRadius: 6,
        backgroundColor: "#FFF",
        shadowColor: "#000000",
        borderWidth: 1,
        borderColor: "#365173",
        shadowOffset: {
            width: 0,
            height: 3
        },
        shadowRadius: 2,
        shadowOpacity: 1
    },
    availableOffline: {
        flexDirection: "row",
        flex: 1,
        alignItems: "center"
    },
    textOffline: {
        color: Res.colors.successColor,
        justifyContent: "flex-start",
        paddingLeft: 5
    }
});
