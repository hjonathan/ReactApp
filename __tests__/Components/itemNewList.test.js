import 'react-native';
import React from 'react';
import ItemNewList from '../../App/Components/ItemNewList';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('Test of new item card for list', () => {

    let store, container;
    const parent = {
            navigation: {
                dispatch: () => {
                },
                state: {
                    routeName: 'Participated'
                }
            }
        },
        initialState = {
            data: {
                autoRoot: "FALSE",
                forms: [{
                    description: "",
                    formId: "",
                    formUpdateDate: "2018-01-18T00:23:52+00:00",
                    index: 1,
                    stepCondition: "",
                    stepId: "",
                    stepMode: "",
                    stepPosition: 1,
                    stepUidObj: "",
                    title: "",
                    triggers: {before: false, after: false}
                }
                ],
                offlineEnabled: "",
                processId: "",
                taskId: "test Task ID",
                text: ""
            },
            progress: ''
        },
        mockStore = configureStore();

    beforeEach(() => {
        store = mockStore(initialState)
        container = shallow(<ItemNewList store={store} parent={parent}/>)
    });

    it('render Item correctly', () => {
        expect(toJson(container)).toMatchSnapshot();
    });

    it('Render the connected(SMART) component', () => {
        expect(container.length).toEqual(1)
    });

    it('Check Prop matches with initialState', () => {
        expect(container.instance().state.data.autoRoot).toEqual('FALSE');
    });

    it('goToFormRender function', () => {
        expect(container.instance().goToFormRender()).toMatchSnapshot();
    });
})