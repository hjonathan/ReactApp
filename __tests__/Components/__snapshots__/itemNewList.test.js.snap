// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of new item card for list goToFormRender function 1`] = `undefined`;

exports[`Test of new item card for list render Item correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  disabled={false}
  onPress={[Function]}
>
  <Component
    elevation={3}
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderColor": "#365173",
        "borderRadius": 6,
        "borderWidth": 1,
        "flex": 1,
        "margin": 3,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderRadius": 6,
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingHorizontal": 10,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../App/Assets/icons/newBlack.png",
            }
          }
          style={
            Object {
              "height": 36,
              "resizeMode": "contain",
              "width": 36,
            }
          }
        />
      </Component>
      <Component
        style={
          Object {
            "paddingVertical": 13,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000",
              "paddingBottom": 3,
            }
          }
        />
      </Component>
    </Component>
  </Component>
</TouchableOpacity>
`;
